### DISCUSSION TITLE
Scalene: Program did not run for long enough to profile.

### DISCUSSION THREAD

**Discussion by Karel-Tavernier** [plasma-umass/scalene#586] [Q&A]
I downloaded the latest version of scalene on Windows 11 with Python 3.11. I ran is from the command line as follows
    C:\Users\kt\OneDrive\Transfer\Python\intersections\repository\src>scalene --cpu --reduced_profile random_test.py
I get the following rather unhelpful message.
       _Error in program being profiled:
       Unknown Error
       Scalene: Program did not run for long enough to profile.
       NOTE: The GPU is currently running in a mode that can reduce Scalene's accuracy when reporting GPU utilization.
       Run once as Administrator or root (i.e., prefixed with `sudo`) to enable per-process GPU accounting._
The script random_test.py runs fine, and takes 1-2 minutes.  This should be long enough, I suppose, and anyhow the error message appears much quicker than that,

I am stuck. And I would like to use scalene, according to the docs it really does what I was looking for.

Any suggestions?

**Comment by emeryberger** [comment]
Any hope you can share the `random_test.py` program? That would help immensely.

**Reply by Karel-Tavernier** [reply]
Sure. Attached are the sources. (They have one dependency, to the gerber_writer PyPI  module.)
[intersections_sources.zip](https://github.com/plasma-umass/scalene/files/11087043/intersections_sources.zip)

Here is what happens when I run the program directly or from scalene.
**### _**C:\Users\kt\OneDrive\Transfer\Python\intersections\repository\src>py random_test.py
Generated 1,000,000 segments, arc fraction 0.60, repeating True in  10.56s
Started to process 1,000,000 segments.
[...]
Segments: 1,000,000, overlapping pairs: 238,227, clock:  43.20s, process:  25.89s, scaled duration: 1.5135, algorithm: find_overlaps_KT
C:\Users\kt\OneDrive\Transfer\Python\intersections\repository\src>scalene --src random_test.py
Error in program being profiled:
 Unknown Error
Scalene: Program did not run for long enough to profile.
NOTE: The GPU is currently running in a mode that can reduce Scalene's accuracy when reporting GPU utilization.
Run once as Administrator or root (i.e., prefixed with `sudo`) to enable per-process GPU accounting.**_**

I guess scalene makes an assumption about the environment that is not fulfilled. Maybe the windows path is useful:
_C:\Users\kt\OneDrive\Transfer\Python\intersections\repository\src>echo %PATH%
C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\GitHub CLI\;C:\Program Files\Git\cmd;C:\Program Files (x86)\Common Files\Isabel\;C:\Program Files\Common Files\Isabel\;C:\Users\kt\AppData\Local\Programs\Python\Python311\Scripts\;C:\Users\kt\AppData\Local\Programs\Python\Python311\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Common Files\Isabel\;C:\Program Files\Common Files\Isabel\;C:\Users\kt\AppData\Local\Microsoft\WindowsApps;;C:\Users\kt\AppData\Local\Programs\Python\Python310\Scripts;C:\Users\kt\AppData\Local\Microsoft\WindowsApps;C:\Users\kt\AppData\Local\Programs\Microsoft VS Code\bin_

Thanks for helping out.

**Comment by Karel-Tavernier** [comment]
Here is a reduced version of the source. I removed the dependency and cut some other parts of the code. The rest is not simple to reduce.
Same Scalene behavior. 

It still throws an "Unknown Error"
[random_test.zip](https://github.com/plasma-umass/scalene/files/11099515/random_test.zip)

**Comment by Karel-Tavernier** [comment]
x

**Comment by Karel-Tavernier** [comment]
I wrote a very simple program, it throws a stack overflow exception in scalene.
I will post this in a separate discussion to simplify things.
